### PR TITLE
Modified the FB API to use the graph API now.

### PR DESCRIPTION
**Pull request by nikhil4globant**
1.Here made changes to make use of new Facebook graph API.

2.Now the graph api returns the expected metrics from facebook (need to modify for more granularity)

### CODE REVIEW

File: spec/fixtures/facebook.json [hunk 1]
```diff
@@ -9,7 +9,8 @@
    },
    "share": {
       "comment_count": 0,
-      "share_count": 9972
+      "share_count": 9972,
```
Comment (mbowen) [inline spec/fixtures/facebook.json]: i think the share_count metric is different from the total even though confusingly called shared_count in the graph api. i think share_count should be 0 in the fixture data.

File: app/models/sources/facebook.rb [hunk 2]
```diff
@@ -2,13 +2,7 @@ class Facebook < Source
   def get_query_url(work, options = {})
     fail ArgumentError, "No Facebook access token." unless get_access_token
     return {} unless work.get_url
-
-    # use depreciated v2.0 API if url_linkstat is used
-    if url_linkstat.present?
-      URI.escape(url_linkstat % { access_token: access_token, query_url: work.canonical_url_escaped })
-    else
       url % { access_token: access_token, query_url: work.canonical_url_escaped }
```
Comment (mbowen) [inline app/models/sources/facebook.rb]: nitpick: maybe move this line to the left 2 places to line up with return. it's not a nested statement.

File: app/models/sources/facebook.rb [hunk 3]
```diff
@@ -19,29 +13,24 @@ def parse_data(result, work, options={})
...
       extra = {}
-    elsif url_linkstat.blank?
-      readers, comments, likes = 0, 0, 0
-      extra = result
     else
-      readers = result.deep_fetch('data', 0, 'share_count') { 0 }
-      comments = result.deep_fetch('data', 0, 'comment_count') { 0 }
-      likes = result.deep_fetch('data', 0, 'like_count') { 0 }
-      extra = result['data'] || {}
+      readers = result.deep_fetch('share', 'share_count') { 0 }
```
Comment (mbowen) [inline app/models/sources/facebook.rb]: share_count in response maps to total which you capture above. I don't think it should be assign to readers. hmm... I think we should maybe hard-code to 0 for now cuz we don't know it (ie, readers = 0).

File: app/models/sources/facebook.rb [hunk 4]
```diff
@@ -19,29 +13,24 @@ def parse_data(result, work, options={})
...
+      likes = result.deep_fetch('share', 0, 'like_count') { 0 }
+      url = result['id']
+      # Need clarification on what to use for getting facebook likes from the graph api
+      extra = {
+               "comment_count" => comments, 
+               "share_count" =>  readers, 
+               "like_count" => likes,
+               "url" => url,
+               "total_count" => total
+      } || {}
```
Comment (mbowen) [inline app/models/sources/facebook.rb]: under what condition would the 2nd expression of logical OR get evaluated? seems like first map expression will always be true.

Comment (nikhil4globant) [inline app/models/sources/facebook.rb]: ah agreed. will change that .

File: app/models/sources/facebook.rb [hunk 5]
```diff
@@ -19,29 +13,25 @@ def parse_data(result, work, options={})
...
-    elsif url_linkstat.blank?
-      readers, comments, likes = 0, 0, 0
-      extra = result
     else
-      readers = result.deep_fetch('data', 0, 'share_count') { 0 }
-      comments = result.deep_fetch('data', 0, 'comment_count') { 0 }
-      likes = result.deep_fetch('data', 0, 'like_count') { 0 }
-      extra = result['data'] || {}
+      readers = 0
+      share_count = result.deep_fetch('share', 'share_count') {0}
```
Comment (nikhil4globant) [inline app/models/sources/facebook.rb]: Hey Matt, 
The share count indeed is less than the total, but in FB graph API the total count is (summation of share_count, likes_count and comments_count). But in our case we are not able to get it down to more granular level. 

the share count is contained in share 
`share {
share_count: 123,
comment_count: 23}`
may be once we get how to do it more granular then we can add all these fields separately. 
That off course is my thought.

**Pull request merged**